File: LeetCode/Easy/0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.go
func strStr(haystack string, needle string) int {
    n := len(haystack)
    m := len(needle)

    if (n < m) {
        return -1
    }

    for i, _ := range(haystack) {
        if (i+m >= n) {
            break
        }
        if (haystack[i:i+m] == needle) {
            return i
        }
    }
    return -1
}

File: LeetCode/Easy/0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.java
class Solution {
    public int strStr(String haystack, String needle) {
        int n = haystack.length();
        int k = needle.length();
        for (int i = 0; i <= n-k; i++) {
            if (haystack.substring(i, i+k).equals(needle)){
                return i;
            }
        }
        return -1;
    }
}